roles/zuul/tasks/main.yml

```diff
@@ -136,6 +136,17 @@
   include: merger.yml
   when: "'zuul-merger' in zuul_components"
 
+- name: Stop unwanted zuul services
+  service:
+    name: "{{ item }}"
+    state: stopped
+    enabled: no
+  when: "item not in zuul_components"
+  with_items:
+    - zuul-launcher
+    - zuul-merger
+    - zuul-server
```

Comment:
If we were to switch to multiple zuul roles, this would probably fall apart. But currently zuul_components is going to be constant on each host.